src/states/elder/mod.rs

```diff
@@ -1057,11 +1061,26 @@ impl Elder {
         new_client_auth: &Authority<XorName>,
         outbox: &mut EventBox,
     ) {
+        #[cfg(feature = "mock_base")]
+        {
+            if self.ignore_candidate_info_counter > 0 {
+                self.ignore_candidate_info_counter -= 1;
+                return;
+            }
+        }
         debug!(
             "{} Handling CandidateInfo from {}->{}.",
             self, old_pub_id, new_pub_id
         );
 
+        if self.peer_mgr.has_candidate_info() {
```

Comment:
I do not think we want that here.
It seem the bug is either in `self.peer_mgr.handle_candidate_info`, or the match processing the result.

When we call this we already know this is a valid candidate we are processing since `self.chain.matching_candidate_target_interval` did not lead us to return early, so we probably should not disconnect it then.

Reply:
I'm confused a bit, so what was the fix for that?

Reply:
This itself is not trying to fix a bug. It's mainly just my over-caustious about the duplicated candidate_info.
I double check the code flow after that point and the case of duplication shall be addressed already.
So I just removed this check here.

Reply:
I forgot and meant to say:
This is a good place for a unit tests. You add a new test that follow the pattern of the existing one to exercise this corner case and verify it does the right thing.